presto-main/src/main/java/com/facebook/presto/execution/ThreadCountParser.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Long.parseLong;
+import static java.lang.Math.multiplyExact;
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+
+// Based on https://github.com/trinodb/trino/blob/5b39de318192c34e08c8e1878fda989c97b8ca57/core/trino-main/src/main/java/io/trino/execution/ThreadCountParser.java
+public class ThreadCountParser
+{
+    private static final String PER_CORE_SUFFIX = "C";
+    private static final Supplier<Integer> AVAILABLE_PROCESSORS = Runtime.getRuntime()::availableProcessors;
+    public static final ThreadCountParser DEFAULT = new ThreadCountParser(AVAILABLE_PROCESSORS);
+
+    private final Supplier<Integer> coreCount;
+
+    @VisibleForTesting
+    ThreadCountParser(Supplier<Integer> coreCount)
+    {
+        this.coreCount = requireNonNull(coreCount, "coreCount is null");
+    }
+
+    public int parse(String value)
+    {
+        int coreCount = this.coreCount.get();
+        checkState(coreCount > 0, "coreCount must be positive");
+
+        long threads;
+        if (value.endsWith(PER_CORE_SUFFIX)) {
+            long multiplier = parseLong(value.substring(0, value.length() - PER_CORE_SUFFIX.length()).trim());
```

Comment:
Do you think there's any benefit in allowing floating point values? e.g. `1.25C`, `1.5C`, etc? It may introduce some additional complexity in that we'd need to round/floor non-integer results, but if you have a large machine, say 48 cores, you would be stuck with only having the option of using 48, 96, or even large multiples with no in-between. It seems like a pretty big jump and I think allowing a floating point value would allow some additional flexibility for those cases